libtransact/src/contract/context/handler.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright 2019 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::hash::Hash;
+
+use crate::contract::address::Addresser;
+use crate::contract::context::key_value::KeyValueTransactionContext;
+use crate::handler::{ApplyError, TransactionContext, TransactionHandler};
+use crate::protocol::transaction::TransactionPair;
+
+pub trait SmartContract: Send {
+    type Key: Eq + Hash;
+    type Addr: Addresser<Self::Key>;
+
+    fn get_family_name(&self) -> &str;
```

Comment:
This trait should not have get_family_name or get_family_verisons (which are used for TP routing only).

Reply:
Ok, those were only present to implement the corresponding TransactionHandler methods.